server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +202,85 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        InteractionNode interactionNode = spanToNode(span);
+        interactionNode.setNodes(recursiveTraceFragment(tenantId, id));
+
+        Trace trace = spanToTrace(span);
+        trace.getNodes().add(interactionNode);
+
+        return trace;
+    }
+
+
+    private List<Node> recursiveTraceFragment(String tenantId, String spanId) {
+        List<Span> spanChildren = getChildren(tenantId, spanId);
+        if (spanChildren == null) {
+            return new LinkedList<>();
+        }
+
+        List<Node> nodes = new ArrayList<>();
+
+        for (Span span: spanChildren) {
+            if (span.serverSpan()) {
```

Comment:
Would still prefer to remove this, and instead guard the call to 'recursiveTraceFragment' below which a !span.clientSpan() - as the children of a client span will only be server spans, so it is a redundant query.